packages/fxa-auth-server/test/scripts/update-subscriptions-to-new-plan.ts

```diff
@@ -0,0 +1,361 @@
+        await subscriptionUpdater.processSubscription(mockFirestoreSub);
+
+        expect(writeReportStub.notCalled).true;
+      });
+
+      it('aborts if account for customer does not exist', async () => {
+        dbStub.account.resolves(null);
+        await subscriptionUpdater.processSubscription(mockFirestoreSub);
+
+        expect(writeReportStub.notCalled).true;
+      });
+
+      it('does not move subscription if subscription is not in active state', async () => {
+        await subscriptionUpdater.processSubscription({
+          ...mockFirestoreSub,
+          status: 'canceled',
+        });
+
+        expect(updateSubscriptionStub.notCalled).true;
+        expect(writeReportStub.notCalled).true;
+      });
+    });
+  });
+
+  describe('fetchCustomer', () => {
+    let customerRetrieveStub: sinon.SinonStub;
+    let result: Stripe.Customer | Stripe.DeletedCustomer | null;
+
+    describe('customer exists', () => {
+      beforeEach(async () => {
+        customerRetrieveStub = sinon.stub().resolves(mockCustomer);
+        stripeStub.customers.retrieve = customerRetrieveStub;
+
+        result = await subscriptionUpdater.fetchCustomer(mockCustomer.id);
+      });
+
+      it('fetches customer from Stripe', () => {
+        expect(
+          customerRetrieveStub.calledWith(mockCustomer.id, {
+            expand: ['subscriptions'],
+          })
+        ).true;
+      });
+
+      it('returns customer', () => {
+        sinon.assert.match(result, mockCustomer);
+      });
+    });
+
+    describe('customer deleted', () => {
+      beforeEach(async () => {
+        const deletedCustomer = {
+          ...mockCustomer,
+          deleted: true,
+        };
+        customerRetrieveStub = sinon.stub().resolves(deletedCustomer);
+        stripeStub.customers.retrieve = customerRetrieveStub;
+
+        result = await subscriptionUpdater.fetchCustomer(mockCustomer.id);
+      });
+
+      it('returns null', () => {
+        sinon.assert.match(result, null);
+      });
+    });
+  });
+
+  describe('updateSubscription', () => {
+    let retrieveStub: sinon.SinonStub;
+    let updateStub: sinon.SinonStub;
+
+    beforeEach(async () => {
+      retrieveStub = sinon.stub().resolves(mockSubscription);
+      stripeStub.subscriptions.retrieve = retrieveStub;
+
+      updateStub = sinon.stub().resolves();
+      stripeStub.subscriptions.update = updateStub;
+
+      await subscriptionUpdater.updateSubscription(mockSubscription);
+    });
+
+    it('retrieves the subscription', () => {
+      expect(retrieveStub.calledWith(mockSubscription.id)).true;
+    });
+
+    it('updates the subscription', () => {
+      expect(
+        updateStub.calledWith(mockSubscription.id, {
+          proration_behavior: prorationBehavior,
+          items: [
+            {
+              id: mockSubscription.items.data[0].id,
+              plan: 'updated',
+            },
+          ],
+        })
```

Comment:
I think this will need the mock metadata for `previous_plan_id` and `plan_change_date`?

Reply:
Indeed!